### PR TITLE
fix(chips): fix chip list focus and keyboard behaviors

### LINKED ISSUE #7240: Chip List incorrectly sets focus to Chip Input when chips are remove programattically
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
When all chips in a chip list are removed programmatically, focus should not be moved to the associated chip input when the current focus is outside the chip list.

#### What is the current behavior?
The chip input is getting focused.

#### What are the steps to reproduce?
1. Add chips to a chip list
2. Tab out and move focus away from chip list and input
3. Clear the chips programmatically (e.g. using a button)

See http://plnkr.co/edit/urO2dB?p=preview

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular: 4.3
Material: Beta 10

### PR DESCRIPTION
**Pull request by tinayuangao**
Fixes #7240 #7242

### CODE REVIEW

File: src/lib/chips/chip-list.ts [hunk 1]
```diff
@@ -432,9 +432,9 @@ export class MdChipList implements MdFormFieldControl<any>, ControlValueAccessor
 
     let isPrevKey = (code === (isRtl ? RIGHT_ARROW : LEFT_ARROW));
     let isNextKey = (code === (isRtl ? LEFT_ARROW : RIGHT_ARROW));
-    let isBackKey = (code === BACKSPACE || code == DELETE || code == UP_ARROW || isPrevKey);
+    let isDeleteKey = code === BACKSPACE || code == DELETE;
     // If they are on an empty input and hit backspace/delete/left arrow, focus the last chip
```
Comment (mmalerba) [inline src/lib/chips/chip-list.ts]: nit: not left arrow anymore

Comment (tinayuangao) [reply src/lib/chips/chip-list.ts]: Updated comment. Thanks for review!

File: src/lib/chips/chip-list.ts [hunk 2]
```diff
@@ -432,9 +432,9 @@ export class MdChipList implements MdFormFieldControl<any>, ControlValueAccessor
 
     let isPrevKey = (code === (isRtl ? RIGHT_ARROW : LEFT_ARROW));
     let isNextKey = (code === (isRtl ? LEFT_ARROW : RIGHT_ARROW));
-    let isBackKey = (code === BACKSPACE || code == DELETE || code == UP_ARROW || isPrevKey);
-    // If they are on an empty input and hit backspace/delete/left arrow, focus the last chip
-    if (isInputEmpty && isBackKey) {
+    let isDeleteKey = code === BACKSPACE || code == DELETE;
+    // If they are on an empty input and hit backspace/delete, focus the last chip
```
Comment (jelbourn) [inline src/lib/chips/chip-list.ts]: I'm not sure the delete key should do this in addition to backspace. In both Drive and GMail, the delete key just does nothing when your focus is on the empty input.

Comment (tinayuangao) [reply src/lib/chips/chip-list.ts]: Removed delete key. Thanks for review!

**Pull request merged**